src/wallet/monero_wallet_light.cpp

```diff
@@ -0,0 +1,3706 @@
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ * Parts of this file are originally copyright (c) 2014-2019, The Monero Project
+ *
+ * Redistribution and use in source and binary forms, with or without modification, are
+ * permitted provided that the following conditions are met:
+ *
+ * All rights reserved.
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this list of
+ *    conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice, this list
+ *    of conditions and the following disclaimer in the documentation and/or other
+ *    materials provided with the distribution.
+ *
+ * 3. Neither the name of the copyright holder nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software without specific
+ *    prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+ * THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+ * THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * Parts of this file are originally copyright (c) 2012-2013 The Cryptonote developers
+ */
+
+#include "wallet/wallet_rpc_server_commands_defs.h"
+#include "monero_wallet_light.h"
+#include "utils/monero_utils.h"
+#include <thread>
+#include <chrono>
+#include <iostream>
+#include "mnemonics/electrum-words.h"
+#include "mnemonics/english.h"
+#include "cryptonote_basic/cryptonote_format_utils.h"
+#include "cryptonote_basic/cryptonote_basic_impl.h"
+#include "string_tools.h"
+#include "device/device.hpp"
+#include "common/threadpool.h"
+#define KEY_IMAGE_EXPORT_FILE_MAGIC "Monero key image export\003"
+
+#define MULTISIG_EXPORT_FILE_MAGIC "Monero multisig export\001"
+
+#define OUTPUT_EXPORT_FILE_MAGIC "Monero output export\004"
+
+#define UNSIGNED_TX_PREFIX "Monero unsigned tx set\005"
+
+using namespace epee;
+using namespace tools;
+using namespace crypto;
+
+/**
+ * Public library interface.
+ */
+namespace monero {
+
+namespace light {
```

Comment:
These functions are also declared and implemented in the monero_wallet_full.cpp file, and therefore I could not import them to monero_wallet_light.cpp, because them are not declared in the monero_wallet_full.h. However, if I tried to copy the implementation without putting namespace, and at the time of compilation I got errors, as those functions were already declared and implemented in monero_wallet_full.cpp. Do you say it would be a good idea to expose the declaration of these functions in that namespace in monero_wallet_full.h? But I think that once monero_wallet_light extends the monero_wallet_full class there will no longer be a need for this namespace.